app-libs/stf/src/stf_sgx.rs

```diff
@@ -182,6 +182,14 @@ impl Stf {
 					Self::shield_funds(who, value)?;
 					Ok(())
 				},
+				TrustedCall::new_game(sender, opponent) => {
+					let origin = sgx_runtime::Origin::signed(sender.clone());
```

Comment:
The origin is not a player but the enclave, so the new_game call should also include the root, which will then be set as the origin.

https://github.com/ajuna-network/ajuna-node/blob/update-substrate-5/pallets/ajuna-connectfour/src/lib.rs#L263